domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -233,6 +229,19 @@ func (pm *pruningManager) deleteBlock(blockHash *externalapi.DomainHash) (alread
 	return false, nil
 }
 
+func (pm *pruningManager) CalculateIndependentPruningPoint(blockHash *externalapi.DomainHash) (*externalapi.DomainHash, error) {
+	ghostdagData, err := pm.ghostdagDataStore.Get(pm.databaseContext, blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	targetBlueScore := uint64(0)
+	if ghostdagData.BlueScore() > pm.pruningDepth {
+		targetBlueScore = ((ghostdagData.BlueScore() - pm.pruningDepth) / pm.finalityInterval) * pm.finalityInterval
```

Comment:
Please write a comment explaining what exactly you are doing (I assume you want to round this to finalityIntervals, but this is far from trivial - so please comment)